homeassistant/components/home_plus_control/api.py

```diff
@@ -0,0 +1,352 @@
+
+        for plant in plants_to_pop:
+            self._plants.pop(plant, None)
+
+        return self._plants
+
+    async def async_handle_module_status(self):
+        """Recover the topology information for the plants defined in this object.
+
+        By requesting the topology of the plant, the system learns about the modules that exist.
+        The topology indicates identifiers, type and other device information, but it contains no
+        information about the state of the module.
+
+        This method returns the list of switch modules that will be registered in HomeAssistant.
+        At this time the switches that are discovered through this API call are flattened into a
+        single data structure.
+
+        Returns:
+            dict: Dictionary of switch modules across all of the plants.
+        """
+        for plant in self._plants.values():
+
+            if self._should_check(
+                CONF_PLANT_TOPOLOGY_UPDATE_INTERVAL,
+                self._refresh_intervals[CONF_PLANT_TOPOLOGY_UPDATE_INTERVAL],
+            ):
+                self.logger.debug(
+                    "API update of plant topology for plant %s.", plant.id
+                )
+                try:
+                    await plant.update_topology()  # Call the API
+                except Exception as err:  # pylint: disable=broad-except
+                    self.logger.error(
+                        "Error encountered when updating plant topology for plant %s: %s [%s]",
+                        plant.id,
+                        err,
+                        type(err),
+                    )
+                else:
+                    # If all goes well, we update the last check time
+                    self._last_check[
+                        CONF_PLANT_TOPOLOGY_UPDATE_INTERVAL
+                    ] = time.monotonic()
+
+            if self._should_check(
+                CONF_MODULE_STATUS_UPDATE_INTERVAL,
+                self._refresh_intervals[CONF_MODULE_STATUS_UPDATE_INTERVAL],
+            ):
+                self.logger.debug("API update of module status for plant %s.", plant.id)
+                try:
+                    await plant.update_module_status()  # Call the API
+                except Exception as err:  # pylint: disable=broad-except
+                    self.logger.error(
+                        "Error encountered when updating plant module status for plant %s: %s [%s]",
+                        plant.id,
+                        err,
+                        type(err),
+                    )
+                else:
+                    # If all goes well, we update the last check time
+                    self._last_check[
+                        CONF_MODULE_STATUS_UPDATE_INTERVAL
+                    ] = time.monotonic()
+
+        return self._update_modules()
+
+    def _should_check(self, check_type, period):
+        """Return True if the current monotonic time is > the last check time plus a fixed period.
```

Comment:
What's the reason we need this advanced time keeping? Could we instead just have two coordinators working with different fixed intervals?